packages/editor-ui/src/composables/useCanvasPanning.ts

```diff
@@ -4,18 +4,32 @@ import { ref, unref } from 'vue';
 import { getMousePosition } from '@/utils/nodeViewUtils';
 import { useUIStore } from '@/stores/ui.store';
 import { useDeviceSupport } from 'n8n-design-system';
-
+import { MOUSE_EVENT_BUTTON, MOUSE_EVENT_BUTTONS } from '@/constants';
+
+/**
+ * Composable for handling canvas panning interactions - it facilitates the movement of the
+ * canvas element in response to mouse events
+ *
+ * @param elementRef Ref pointing to the canvas HTML element that will be panned
+ * @param options Optional settings
+ * @param options.onMouseMoveEnd Callback function ref that will be called when the mouse movement ends
+ */
 export function useCanvasPanning(
 	elementRef: Ref<null | HTMLElement>,
 	options: {
-		// To be refactored (unref) when migrating NodeView to composition API
+		// @TODO To be refactored (unref) when migrating NodeView to composition API
 		onMouseMoveEnd?: Ref<null | ((e: MouseEvent) => void)>;
 	} = {},
 ) {
 	const uiStore = useUIStore();
 	const moveLastPosition = ref([0, 0]);
 	const deviceSupport = useDeviceSupport();
 
+	/**
+	 * Updates the canvas offset position based on the mouse movement
+	 *
+	 * @param e The mouse event
```

Comment:
Nice job adding documentation 🎉 

One optional suggestion to improve clarity. IMHO just repeating the param in the JSDoc without extra info, or repeating the param name in different words is a bit extra noise. E.g. here it's pretty obvious what the param is, so maybe it could be omitted. Same comment about the JSDoc block.

```suggestion
```